src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -0,0 +1,371 @@
+        """Create a `RelaxedRigidContactsState` instance"""
+
+        return RelaxedRigidContactsState()
+
+    @staticmethod
+    def zero(model: js.model.JaxSimModel) -> RelaxedRigidContactsState:
+        """Build a zero `RelaxedRigidContactsState` instance from a `JaxSimModel`."""
+        return RelaxedRigidContactsState.build()
+
+    def valid(self, model: js.model.JaxSimModel) -> bool:
+        return True
+
+
+@jax_dataclasses.pytree_dataclass
+class RelaxedRigidContacts(ContactModel):
+    """Rigid contacts model."""
+
+    parameters: RelaxedRigidContactsParams = dataclasses.field(
+        default_factory=RelaxedRigidContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    def compute_contact_forces(
+        self,
+        position: jtp.Vector,
+        velocity: jtp.Vector,
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+    ) -> tuple[jtp.Vector, tuple[Any, ...]]:
+
+        def _detect_contact(x: jtp.Array, y: jtp.Array, z: jtp.Array) -> jtp.Array:
+            x, y, z = jax.tree_map(jnp.squeeze, (x, y, z))
+
+            n̂ = self.terrain.get_normal_at(x=x, y=y).squeeze()
+            h = jnp.array([0, 0, z - model.terrain.get_height_at(x=x, y=y)])
+
+            return jnp.array([0, 0, jnp.dot(h, n̂)], dtype=float)
+
+        # Compute the activation state of the collidable points
+        position = jax.vmap(_detect_contact)(*position.T)
+
+        with data.switch_velocity_representation(VelRepr.Mixed):
+            M = js.model.free_floating_mass_matrix(model=model, data=data)
+            J_WC = jnp.vstack(
+                jax.vmap(lambda j, height: j * (height < 0))(
+                    js.contact.jacobian(model=model, data=data)[:, :3], position[:, 2]
+                )
+            )
+            W_H_C = js.contact.transforms(model=model, data=data)
+            h = js.model.free_floating_bias_forces(model=model, data=data)
+            W_ν = data.generalized_velocity()
+            J̇_WC = jnp.vstack(
+                jax.vmap(lambda j, height: j * (height < 0))(
+                    js.contact.jacobian_derivative(model=model, data=data)[:, :3],
+                    position[:, 2],
+                ),
+            )
+
+            a_ref, R = self._regularizers(
+                model=model,
+                position=position,
+                velocity=velocity,
+                parameters=self.parameters,
+            )
+
+        G = J_WC @ jnp.linalg.lstsq(M, J_WC.T)[0]
+
+        CW_al_free_WC = J_WC @ jnp.linalg.lstsq(M, -h)[0] + J̇_WC @ W_ν
+
+        # Calculate quantities for the linear optimization problem.
+        A = G + R
+        b = CW_al_free_WC - a_ref
+
+        objective = lambda x: jnp.sum(jnp.square(A @ x + b))
+
+        # Compute the 3D linear force in C[W] frame
+        opt = jaxopt.LBFGS(
+            fun=objective,
+            maxiter=50,
+            tol=1e-6,
```

Comment:
I'd probably expose also (at least) these two as parameters as they are likely related to the trade-off between speed and accuracy of the contact model.

Reply:
Absolutely! Thanks

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/223/commits/2ea87526ad4f100a90285a29674b4bd7df0cadf2